### ISSUE TITLE
PlaybackSpeed, Long-running processes, Weird non-blocking behaviour

### ISSUE THREAD

**Issue by mattvonrocketstein** [charmbracelet/vhs#481]
**Describe the bug**

Like the title says, I'm seeing some weirdness with PlaybackSpeed, long-running processes, and non-blocking behaviour.  Honestly not sure if this is a request for help, a bug report, or a feature request.

**Setup**

Ubuntu / bash / gnome-terminal, vhs version v0.7.2 (484e440)

**To Reproduce**

Sleep a long time to simulate a long-running command, and try to speed up the recap in the recorded gif.

**Source Code**

Here's an example test.tape:
 
```
Output img/test.gif 
Set Shell "bash"
Set FontSize 22
Set Width 1200
Set Height 1000
Set TypingSpeed 100ms
Type "sleep 2; echo done sleeping 2s"
Enter
Sleep 3
Set PlaybackSpeed 100
Type "sleep 100; echo done sleeping 100s"
Enter
Sleep 3
Type "echo doing next thing"
Enter
Sleep 3
Type "finishing immediately"
Enter
```
**Expected behavior**

If I sleep 100s and have 100x playback, I'd expect the recording to show a pause of 1s and then show "done sleeping 100s".
I'd also expect that vhs takes at least 100s to record the video, because it has to record in real time even if the playback is sped up.

**Actual behaviour**

3s into the 100s sleep, "done" prints, as if vhs got bored waiting on the first longer sleep.

![test](https://github.com/charmbracelet/vhs/assets/182887/440ca174-0200-4e19-a4bd-6c48118cae9b)

Based on the output from `vhs test.tape` during generation of the above, it looks like it indeed just skips to the next steps in the middle of the 100s sleep, but shouldn't these be blocking steps that run in series?

**Additional context**

My use case involves recording a terminal session where there are several commands, some returning in seconds and some taking several minutes.  Fast things don't need to be faster, but the slow things do actually need a *large* speedup because most of the time is spent waiting and there's not screen changes (ie just polling a cluster until it's ready).  I can't know in advance exactly how long it will take for the commands to run, so adjusting timing pauses  and playback speed doesn't work very well.

Rather than adjusting playback speed based on multipliers like 1x or 100x  a base speed, what might be much more natural is to allow the user to set a maximum playback length, then let the system determine the framerate/playback speed that works.

I'm probably doing this wrong.  None of the examples in-repo seem to involve long-running commands, much less commands with kind of randomized run-times, but that seems like it would be a common use-case.

**Comment by NfNitLoop** [comment]
New user, came across your post while looking for a different issue.

Your problem is that you're telling your *shell* to `sleep 100` by typing that to it.

But `vhs` only knows that that you *typed* that. Maybe you typed it into a text editor that's not actually going to sleep. `vhs` doesn't have any idea.

On your next lines you `Enter` and then `Sleep 3`.  You would need to either `Sleep 100` (or maybe a bit more), or `Wait` for the message you expect to appear when you're done sleeping.

**Comment by mattvonrocketstein** [comment]
You are correct, now that I'm less of a new user myself I did eventually figure this out.  Definitely not an application bug per se, but perhaps a docs bug and/or a feature request?  

Details about the concurrency model for vhs instruction set isn't very clear at first glance.  And block-until-exit in the way I initially expected does seem like a useful feature.. otherwise for multi-command recordings one spends a lot of time in a looping workflow where you first guess about execution time, re-record with appropriate sleeps, modify sleeps, repeat
